From 0d2a2afb34ef16d84e4a12b535199651fc48ab68 Mon Sep 17 00:00:00 2001
From: Ming Ma <mingma@apache.org>
Date: Mon, 16 Nov 2015 09:32:40 -0800
Subject: [PATCH 1750/2848] HDFS-9413. getContentSummary() on standby should
 throw StandbyException. (Brahma Reddy Battula via
 mingma)

(cherry picked from commit 02653add98f34deedc27f4da2254d25e83e55b58)
(cherry picked from commit 42b55ff23e5fa30e00a6b56df92777a1899a1952)

(cherry picked from commit 83200450f67c1fe2b17385ade23051d83c7354bf)

(cherry picked from commit 1b0f277eae40b1cf321d0bcca7ed036dbea08b30)

Conflicts:
	hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/namenode/FSNamesystem.java

Change-Id: I3cbdb47b3dfc97bd5ecd97507319eb75b702df26
---
 .../hdfs/server/namenode/ha/TestQuotasWithHA.java  |   15 +++++++++++++++
 1 file changed, 15 insertions(+)

diff --git a/hadoop-hdfs-project/hadoop-hdfs/src/test/java/org/apache/hadoop/hdfs/server/namenode/ha/TestQuotasWithHA.java b/hadoop-hdfs-project/hadoop-hdfs/src/test/java/org/apache/hadoop/hdfs/server/namenode/ha/TestQuotasWithHA.java
index 6ceecc7..db67d95 100644
--- a/hadoop-hdfs-project/hadoop-hdfs/src/test/java/org/apache/hadoop/hdfs/server/namenode/ha/TestQuotasWithHA.java
+++ b/hadoop-hdfs-project/hadoop-hdfs/src/test/java/org/apache/hadoop/hdfs/server/namenode/ha/TestQuotasWithHA.java
@@ -17,6 +17,7 @@
  */
 package org.apache.hadoop.hdfs.server.namenode.ha;
 
+
 import static org.junit.Assert.assertEquals;
 
 import java.io.IOException;
@@ -34,6 +35,7 @@
 import org.apache.hadoop.hdfs.MiniDFSNNTopology;
 import org.apache.hadoop.hdfs.server.namenode.NameNode;
 import org.apache.hadoop.io.IOUtils;
+import org.apache.hadoop.ipc.StandbyException;
 import org.junit.After;
 import org.junit.Before;
 import org.junit.Test;
@@ -130,4 +132,17 @@ public void testQuotasTrackedOnStandby() throws Exception {
     assertEquals(1, cs.getDirectoryCount());
     assertEquals(0, cs.getFileCount());
   }
+
+  /**
+   * Test that getContentSummary on Standby should should throw standby
+   * exception.
+   */
+  @Test(expected = StandbyException.class)
+  public void testgetContentSummaryOnStandby() throws Exception {
+    Configuration nn1conf =cluster.getConfiguration(1);
+    // just reset the standby reads to default i.e False on standby.
+    HAUtil.setAllowStandbyReads(nn1conf, false);
+    cluster.restartNameNode(1);
+    cluster.getNameNodeRpc(1).getContentSummary("/");
+  }
 }
-- 
1.7.9.5

